plugin/metrics/prometheus/metricsstore/reader_test.go

```diff
@@ -25,17 +25,28 @@ import (
 	"github.com/jaegertracing/jaeger/storage/metricsstore"
 )
 
-func TestNewMetricsReader(t *testing.T) {
+const defaultTimeout = 30 * time.Second
+
+func TestNewMetricsReaderValidAddress(t *testing.T) {
 	logger := zap.NewNop()
-	reader, err := NewMetricsReader(logger, nil, time.Second)
+	reader, err := NewMetricsReader(logger, "localhost:1234", defaultTimeout)
 	assert.NoError(t, err)
 	assert.NotNil(t, reader)
 }
 
+func TestNewMetricsReaderInvalidAddress(t *testing.T) {
+	logger := zap.NewNop()
+	reader, err := NewMetricsReader(logger, "\n", defaultTimeout)
+	const wantErrMsg = `parse "http://\n": net/url: invalid control character in URL`
```

Comment:
it's better not to depend on exact error messages from stdlib, but wrap the `error.Wrap` the error with our own message and assert only on our own string, e.g.:
```
require.Error(t, err)
assert.Contains(t, expectedMsg, err.Error())
```

Reply:
yup, I'll address this in the next PR, thanks!